mera/mera_pymer4.py

```diff
@@ -1,10 +1,50 @@
-from typing import List, Optional
-
+import natsort
 import numpy as np
 import pandas as pd
+from typing import List, Optional, NamedTuple
+
 from pymer4.models import Lmer
 
 
+class MeraResults(NamedTuple):
+    """
+    The results of the mixed-effects regression analysis.
+
+    Attributes
+    ----------
+    event_res_df: dataframe
+        Contains the random effect for each event (rows) and IM (columns)
+    event_cond_std_df: dataframe
+        Contains the conditional standard deviations
+        for each event (rows) and IM (columns) (similar to a standard error)
+    rem_res_df: dataframe
+        Contains the leftover residuals for
+        each record (rows) and IM (columns)
+    bias_std_df: dataframe
+        Contains bias, between-event sigma (tau),
+        between-site sigma (phi_S2S) (only when compute_site_term is True),
+        remaining residual sigma (phi_w) and total sigma (sigma) (columns)
+        per IM (rows)
+    fit_df: dataframe
+        Contains the fitted values for each record (rows) and IM (columns)
+    site_res_df: dataframe
+        Contains the random effect for each site (rows) and IM (columns)
+        Note: Only returned if compute_site_term is True
+    site_cond_std_df: dataframe
+        Contains the conditional standard deviations
+        for each site (rows) and IM (columns) (similar to a standard error)
+        Note: Only returned if compute_site_term is True
+    """
+
+    event_res_df: pd.DataFrame
+    event_cond_std_df: pd.DataFrame
+    rem_res_df: pd.DataFrame
+    bias_std_df: pd.DataFrame
+    fit_df: pd.DataFrame
+    site_res_df: Optional[pd.DataFrame]
+    site_cond_std_df: Optional[pd.DataFrame]
+
+
 def run_mera(
     residual_df: pd.DataFrame,
     ims: List[str],
```

Comment:
No need for `List` anymore.
```suggestion
    ims: list[str],
```